src/NServiceBus.Core/Encryption/ConfigureRijndaelEncryptionService.cs

```diff
@@ -169,12 +169,14 @@ IList<byte[]> expiredKeys
         public static void RegisterEncryptionService(this EndpointConfiguration config, Func<IEncryptionService> func)
         {
             Guard.AgainstNull(nameof(config), config);
-            config.Settings.Set("EncryptionServiceConstructor", func);
+
+            config.EnableFeature<Encryptor>();
```

Comment:
Is this more compliant to other features? Then let's do it

On Tuesday, August 2, 2016, Tim Bussmann <notifications@github.com> wrote:

> In src/NServiceBus.Core/Encryption/ConfigureRijndaelEncryptionService.cs
> <https://github.com/Particular/NServiceBus/pull/3991#discussion_r73153659>
> :
>
> > @@ -169,12 +169,14 @@ IList<byte[]> expiredKeys
> >          public static void RegisterEncryptionService(this EndpointConfiguration config, Func<IEncryptionService> func)
> >          {
> >              Guard.AgainstNull(nameof(config), config);
> > -            config.Settings.Set("EncryptionServiceConstructor", func);
> > +
> > +            config.EnableFeature<Encryptor>();
>
> instead of manually enabling the feature on the settings which trigger
> activation, couldn't we check for the setting value on the feature setup as
> a prerequisite and let the feature disable itself?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Particular/NServiceBus/pull/3991/files/12adb8c3c3e5ff39ca90e787eb2944f8bbf3c1ef#r73153659>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAKosiSbqSDbuYTWQCVgjqOBpV_ptNjrks5qb0VEgaJpZM4JYL5j>
> .
>